### PR TITLE
Fixbug int opcode can't start with 0x

### PR DESCRIPTION
**Pull request by vuvth**
## Proposed Changes

+ Support prefix 0x(hex) and 0(oct) for `Int` opcode
+ Add assertNumber for assert data is number with 3 format dec, oct, hex.
+ Unit tests.

### CODE REVIEW

File: packages/runtime/src/lib/parsing.ts [hunk 1]
```diff
@@ -21,6 +21,26 @@ export function assertOnlyDigits (val: string, line: number): void {
...
+ * return val if format is correct
+ * @param val : string
+ */
+export function assertNumber (val: string, line: number): string {
+  if (reOct.test(val)) {
+    // typescript use 0o postfix instade of 0 postfix for oct format.
+    return "0o".concat(val.substring(1));
+  }
+
+  if (reDec.test(val) || reHex.test(val)) return val;
```
Comment (robert-zaremba) [inline packages/runtime/src/lib/parsing.ts:35]: before returning, should we try to parse the number and check if it's smaller than `1<<64` (2**64)?

File: packages/runtime/test/src/parser/parsing.ts [hunk 2]
```diff
@@ -84,6 +84,39 @@ describe("Get Encoding for Byte Data", () => {
   });
 });
 
+describe("assertNumber", function () {
+  it("should pass with hex and dec", () => {
+    const hexValue = "0xaa3C3";
+    assert.deepEqual(assertNumber(hexValue, 1), hexValue);
+    const decValue = "343434";
+    assert.deepEqual(assertNumber(decValue, 1), decValue);
```
Comment (robert-zaremba) [inline packages/runtime/test/src/parser/parsing.ts:92]: don't need to use `deepEqual`. Simple equal is perfect here

**Review by robert-zaremba — APPROVED** [review]
pre-approving

File: packages/runtime/src/lib/parsing.ts (hunk 1 above)
Comment (vuvth) [reply packages/runtime/src/lib/parsing.ts:35]: We check it in Int opcode [here](https://github.com/scale-it/algo-builder/blob/dc544299516315c33ce0269d01fcbe18decbc68f/packages/runtime/src/interpreter/opcode-list.ts#L2154)

File: packages/runtime/test/src/parser/parsing.ts (hunk 2 above)
Comment (vuvth) [reply packages/runtime/test/src/parser/parsing.ts:92]: Thanks!

File: packages/runtime/src/lib/parsing.ts [hunk 3]
```diff
@@ -21,6 +21,26 @@ export function assertOnlyDigits (val: string, line: number): void {
   }
 }
 
+/**
+ * assert if string is valid number respesent.
```
Comment (ratik21) [inline packages/runtime/src/lib/parsing.ts:25]:
```suggestion
 * assert if string is valid algorand number respesentation (octal / hex / unsigned int).
```

**Review by ratik21 — COMMENTED** [review]
We need changelog update

**Pull request merged**